From: Jesse Barnes <jbarnes@virtuousgeek.org>
Date: Wed, 20 Aug 2008 21:40:29 +0000 (-0700)
Subject: Don't allocate a pipe for hotplug detection
X-Git-Url: http://gitweb.freedesktop.org/?p=xorg/driver/xf86-video-intel.git;a=commitdiff;h=7b6f4d22211d71480caf6335a3eacaacff369371

Don't allocate a pipe for hotplug detection

It shouldn't be needed...
---

--- a/src/i830_crt.c
+++ b/src/i830_crt.c
@@ -352,10 +352,9 @@ i830_crt_detect(xf86OutputPtr output)
     xf86OutputStatus	    status;
     Bool		    connected;
 
-    crtc = i830GetLoadDetectPipe (output, NULL, &dpms_mode);
-    if (!crtc)
-	return XF86OutputStatusUnknown;
-
+    /*
+     * Try hotplug detection where supported
+     */
     if (IS_I945G(pI830) || IS_I945GM(pI830) || IS_I965G(pI830) ||
 	    IS_G33CLASS(pI830)) {
 	if (i830_crt_detect_hotplug(output))
@@ -363,12 +362,19 @@ i830_crt_detect(xf86OutputPtr output)
 	else
 	    status = XF86OutputStatusDisconnected;
 
-	goto out;
+	goto done;
     }
 
+    /*
+     * DDC is next best, no flicker
+     */
+    crtc = i830GetLoadDetectPipe (output, NULL, &dpms_mode);
+    if (!crtc)
+	return XF86OutputStatusUnknown;
+
     if (i830_crt_detect_ddc(output)) {
 	status = XF86OutputStatusConnected;
-	goto out;
+	goto out_release_pipe;
     }
 
     /* Use the load-detect method if we have no other way of telling. */
@@ -378,9 +384,10 @@ i830_crt_detect(xf86OutputPtr output)
     else
 	status = XF86OutputStatusDisconnected;
 
-out:
+out_release_pipe:
     i830ReleaseLoadDetectPipe (output, dpms_mode);
 
+done:
     return status;
 }
 
